### DISCUSSION TITLE
Extra line items on cart item for customisable products (variable pricing)

### DISCUSSION THREAD

**Discussion by edinskeja** [medusajs/medusa#4107] [Feature Requests]
I work with a 3D product customizer where you can choose many alternatives on a product for example colors etc. And depending on choices there's a price adjustment. Right now there's no way to manipulate the price in the cart/order and a solution for this would be either `cart line adjustment` or a `extra` array in the `medusa.carts.lineItems.create` function. Here's an example below.

```medusa.carts.lineItems.create('cartId', {
    variant_id: 1,
    quantity: 1,
    extra: [
      {
        hat_color: values.hat_color,
        price: 100
      },
      ...
    ]
  });```

**Comment by burtondev** [comment]
Did you ever find a solution for your custom products?

I started a new discussion called [Does (or Can) Medusa Support Custom Pricing for Customizable Products? #8571](https://github.com/medusajs/medusa/discussions/8571) because I'm not sure a line item is the right solution.
